org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -537,6 +537,16 @@ public class Preferences {
 	 */
 	public static final String CHAIN_COMPLETION_KEY = "java.completion.chain.enabled";
 
+	/**
+	 * Preference key to set the scope value to use when searching java code. Allowed value are
+	 * <ul>
+	 * <li><code>main</code>			-	Scope for main code</li>
+	 * <li><code>all</code>				-	Scope for both test and main code</li>
+	 * </ul>
+	 * Any other unknown value will be treated as <code>all</code>.
+	 */
+	public static final String JAVA_SEARCH_SCOPE = "java.search.scope";
```

Comment:
We could try our best to respect this but there will always be some kinds of searches that don't due to performance/etc. (eg. code lens references won't show more than source references, workspace symbols for now don't show all methods within libraries, etc.)

Reply:
Isn't source reference includes both main and test code in these search scenarios already ? The filtration of test code happens based on a special flag thats available in the classpathentry, but any classpathentry marked as source paths, that contains source code are considered sources If I'm correctly understands those constants.